File: typescript/helpers/find_prime.ts
// Returns the smallest prime p such that p ≡ 1 mod n
import { primeSieve } from '../helpers/primeSieve';
export function find_prime_congruent_one_mod_n(n: number): number {
	let k = 1;
	while (true) {
		const p = k * n + 1;
		const primes = primeSieve(p + 1);
		if (!primes[p]) return p;
		k++;
	}
}

File: java/helpers/FindPrime.java
package helpers;

import java.util.Arrays;
import specialized.primeSieve;

public class FindPrime {

    // Find the smallest prime p such that p ≡ 1 mod n
    public static int findPrimeCongruentOneModN(int n) {
        int k = 1;
        while (true) {
            int p = k * n + 1;
            boolean[] primes = primeSieve.sieve(p+1);
            if (!primes[p]) {
                return p;
            }
            k++;
        }
    }

    // Example usage
    public static void main(String[] args) {
        int n = 7;
        int p = findPrimeCongruentOneModN(n);
        System.out.println("Smallest prime ≡ 1 mod " + n + ": " + p);
    }
}
